rust/frontend/src/impls/service_based_frontend.rs

```diff
@@ -139,6 +140,18 @@ impl ServiceBasedFrontend {
         self.executor.get_supported_segment_types()
     }
 
+    pub async fn get_cached_collection(
```

Comment:
don't think we should address this but the design of this is a bit clunky but i think its OK. Roundtripping into the downstream FE impl for cache access smells a bit off.

Reply:
Agreed, but it felt the least clunky of the options (bringing the provider up a layer or moving the auth)